docs/security.md

```diff
@@ -258,3 +258,51 @@ If you want to use the command `curl`, you can follow the commands:
 openssl x509 -inform der -in $JAVA_HOME/localhost.crt -out certificate.pem
 curl -v -X GET --cacert ./certificate.pem -H "Accept: application/vnd.gravitino.v1+json" -H "Content-Type: application/json" https://localhost:8433/api/version
 ```
+
+## Custom filters
+
+Gravitino supports custom filters to implement the user specified logic to satisfy different safety needs.
+
+### Gravitino server's configuration
+| Configuration item                          | Description                                                      | Default value | Required  | Since version |
+|---------------------------------------------|------------------------------------------------------------------|---------------|-----------|---------------|
+| `gravitino.server.webserver.customFilters`  | Comma separated list of filter class names to apply to the APIs. | ``            | Yes       | 0.4.0         |
+The filter should be a standard javax servlet Filter.
+Filter parameters can also be specified in the configuration, by setting config entries of the form `gravitino.server.webserver.<class name of filter>.param.<param name>=<value>`
+
+For example:
+```text
+gravitino.server.webserver.customFilters=com.test.filter1
+gravitino.server.webserver.com.test.filter1.param.name1=foo
+gravitino.server.webserver.com.test.filter1.param.name2=ba
+```
+
+### Iceberg REST service's configuration
+| Configuration item                                  | Description                                                        | Default value | Required  | Since version |
+|-----------------------------------------------------|--------------------------------------------------------------------|---------------|-----------|---------------|
+| `gravitino.auxService.iceberg-rest.customFilters`   | Comma separated list of filter class names to apply to the APIs.   | ``            | Yes       | 0.4.0         |
```

Comment:
Why is the default value `` and it's a required value?

Reply:
Require value means that the value can't be null. It's ok to use `` as the value. It means that we don't use any filter.

Reply:
I have the same question, `Required` means user must fillup this field or the program couldn't startup as I understand.

Reply:
Required means that this field can't be `null`.

Reply:
> I have the same question, `Required` means user must fillup this field or the program couldn't startup as I understand.

According to your standard, `gravitino.auxService.iceberg-rest.catalog-backend` isn't a required field. But you choose it as a required field.

Reply:
Following the previous document specification, we use "(none)" to indicate no default value.

JDBC catalog:
<img width="1140" alt="image" src="https://github.com/datastrato/gravitino/assets/24897598/82ac569f-7f28-4f9d-a039-d213c766accc">

Hive catalog:
<img width="1147" alt="image" src="https://github.com/datastrato/gravitino/assets/24897598/6d4ab3cd-3062-4ce4-b850-a6a18d33b6cc">

Reply:
OK, I see other documents follow this standard. I will change this.

Reply:
https://github.com/datastrato/gravitino/pull/1214 I raised another pr to fix the comments in other places.
If this config option has default value, this config option is not required.

Reply:
Fixed.